plugins/woocommerce/src/Internal/Admin/Settings/PaymentProviders/WooPayments.php

```diff
@@ -155,38 +156,50 @@ public function get_onboarding_url( WC_Payment_Gateway $payment_gateway, string
 			return add_query_arg( $params, $connect_url );
 		}
 
-		// We don't have an account yet, so the onboarding link is to kickstart the process.
+		// We don't have an account yet, so the onboarding link is used to kickstart the process.
 
-		// Apply our routing logic to determine if we should do a live onboarding/account.
+		// Default to test-account-first onboarding.
 		$live_onboarding = false;
 
-		$onboarding_profile = get_option( OnboardingProfile::DATA_OPTION, array() );
-
 		/*
-		 * For answers provided in the onboarding profile, we will only consider live onboarding if:
-		 * Merchant selected “I’m already selling” and answered either:
-		 * - Yes, I’m selling online.
-		 * - I’m selling both online and offline.
+		 * Apply our routing logic to determine if we should do a live onboarding/account.
+		 *
+		 * For new stores (not yet launched aka in Coming Soon mode),
+		 * based on the answers provided in the onboarding profile, we will do live onboarding if:
+		 * - Merchant selected “I’m already selling” AND answered either:
+		 *   - Yes, I’m selling online.
+		 *   - I’m selling both online and offline.
 		 *
-		 * For existing stores, we will only consider live onboarding if all are true:
+		 * For launched stores, we will only consider live onboarding if all are true:
 		 * - Store is at least 90 days old.
 		 * - Store has an active payments gateway (other than WooPayments).
 		 * - Store has processed a live electronic payment in the past 90 days (any gateway).
 		 *
 		 * @see plugins/woocommerce/client/admin/client/core-profiler/pages/UserProfile.tsx for the values.
 		 */
-		if (
-			isset( $onboarding_profile['business_choice'] ) && 'im_already_selling' === $onboarding_profile['business_choice'] &&
-			isset( $onboarding_profile['selling_online_answer'] ) && (
-				'yes_im_selling_online' === $onboarding_profile['selling_online_answer'] ||
-				'im_selling_both_online_and_offline' === $onboarding_profile['selling_online_answer']
-			)
+		if ( filter_var( get_option( 'woocommerce_coming_soon' ), FILTER_VALIDATE_BOOLEAN ) ) {
```

Comment:
This improves the router logic to only target stores in coming soon mode when determining the flow for new stores.
If the store is not in coming soon mode (if was launched), then we apply the existing stores logic.